### ISSUE TITLE
Allow the graphical `MacosTimePicker` to be updated by dragging the clock hands to a different position

### ISSUE THREAD

**Issue by GroovinChip** [macosui/macos_ui#215]
The native graphical time picker has this functionality, so it should be implemented here.

**Comment by GroovinChip** [comment]
Been thinking about how to do this and I am totally stumped :|

**Comment by whiplashoo** [comment]
Can be probably done with the [touchable](https://pub.dev/packages/touchable) package. I have used it on another project to all kinds of gesture detection to `CustomPaint` shapes.

**Comment by GroovinChip** [comment]
Fascinating!! I will take a close look at this either this evening or tomorrow. Thank you!

**Comment by GroovinChip** [comment]
@whiplashoo I've just experimented with the touchable and I was unable to achieve any results :/

**Comment by whiplashoo** [comment]
Yes, I also had a go and found it will need a lot of work to get right. I managed to add gesture detectors (onPanUpdate) to the clock hands. But then, I got stuck on calculating the gesture delta and properly updating the hand's position.

**Comment by GroovinChip** [comment]
Yeah it's quite a challenge. I'm really not sure what to do :/

On a more concerning note, I found several bugs with the actual time. I'll log those when I get a chance.
